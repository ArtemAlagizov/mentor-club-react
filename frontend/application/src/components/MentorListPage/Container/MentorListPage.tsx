import React, { useEffect } from "react";
import { connect } from "react-redux";
import MentorListPageView from "../View/MentorListPageView";
import { fetchMentorListBegin } from "../../../state/MentorRequests/actions";
import { IAppState } from "../../../store";
import {
  selectIsFetchingMentors,
  selectMentors
} from "../../../state/MentorRequests/selectors";
import { useHistory } from "react-router";
import { IMentorListPageProps } from "./IMentorListPageProps";

export const MentorListPage: React.FC<IMentorListPageProps> = ({
  mentors,
  fetchMentorListBegin,
  isFetchingMentorList
}) => {
  const history = useHistory();

  useEffect(() => {
    fetchMentorListBegin();
  }, [fetchMentorListBegin]);

  const handleChooseMentorClick = (mentorUserId: string) => {
    history.push({
      pathname: `/mentor-requests/new`,
      state: { detail: mentorUserId }
    });
  };

  return (
    <MentorListPageView
      mentors={mentors}
      isFetchingMentorList={isFetchingMentorList}
      handleChooseMentorClick={handleChooseMentorClick}
    />
  );
};

export const mapStateToProps = (state: IAppState) => ({
  mentors: selectMentors(state),
  isFetchingMentorList: selectIsFetchingMentors(state)
});

export const mapDispatchToProps = {
  fetchMentorListBegin
} as const;

export default connect(mapStateToProps, mapDispatchToProps)(MentorListPage);
