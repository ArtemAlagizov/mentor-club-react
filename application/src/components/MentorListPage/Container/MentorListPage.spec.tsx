import React from "react";
import { shallow } from "enzyme";
import { MentorListPage } from "./MentorListPage";

jest.mock("react-router-dom");
jest.mock("react-router", () => ({
  useHistory: () => ({
    push: jest.fn()
  })
}));

describe("Given MentorListPage", () => {
  it("should have MentorListPage matching snapshot", () => {
    const component = shallow(
      <MentorListPage
        mentors={[]}
        fetchMentorListBegin={jest.fn()}
        isFetchingMentorList={false}
      />
    );

    expect(component).toMatchSnapshot();
  });
});
