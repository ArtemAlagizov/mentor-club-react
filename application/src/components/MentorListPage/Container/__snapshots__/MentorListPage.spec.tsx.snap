// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given MentorListPage should have MentorListPage matching snapshot 1`] = `
<MentorListPageView
  handleChooseMentorClick={[Function]}
  isFetchingMentorList={false}
  mentors={Array []}
/>
`;
